lib/tasks/unpublish.rake

```diff
@@ -0,0 +1,42 @@
+require "services"
+
+namespace :unpublish do
+  desc "Unpublish a Finder and redirect the finder page to specific URL"
+  task :redirect_finder, %i[finder_slug redirect_url] => :environment do |_, args|
+    schema_file = Dir.glob("lib/documents/schemas/*.json").find do |file|
+      File.foreach(file).grep(/"base_path": "\/#{args.finder_slug}"/).any?
+    end
+
+    if schema_file.nil?
+      puts "Could not find any finders with that slug. Please check again."
+      next
+    end
+
+    schema = MultiJson.load(File.read(schema_file))
```

Comment:
I think you might be doing things the 'hard' way here. The [publishing_api:publish_finder][1] task makes use of [FinderLoader][2] which seems a simpler way to do it.

I think something like this should work:

```suggestion
  task :redirect_finder, %i[name redirect_url] => :environment do |_, args|
    begin
      schema = FinderLoader.new.finder(args.name)[0].fetch(:file)
    rescue StandardError => e
      puts "Error: #{e.inspect}"
      next
    end
```

[1]: https://github.com/alphagov/specialist-publisher/blob/caf19121fa3a82eec4f0ff828b88a075e09fe275/lib/tasks/publishing_api.rake#L18-L37
[2]: https://github.com/alphagov/specialist-publisher/blob/caf19121fa3a82eec4f0ff828b88a075e09fe275/app/lib/finder_loader.rb

Reply:
The problem with this is, it loads via filename, and not the slug - which is not one and the same. from a support perspective i just thought finding by slug would be the easiest since we can copy from the URL. I could also add the functionality to FinderLoader - what do you think @ollietreend ?

Reply:
@minhngocd I think we should try to make this Rake task consistent with the other one – otherwise, it could be more confusing if the command to re-publish a finder requires a different param than the one to unpublish it.

I don't have a strong opinion about whether the `FinderLoader` should support finding by slug as well as (or instead of) finding by the JSON file name. Given that's how the existing `publishing_api:publish_finder` task works, I'd be inclined to say it isn't a problem – or at least it hasn't been inconvenient enough to have triggered someone to change it already. And since these Rake tasks aren't things we need to run every day, I feel like it's probably fine.

However, if you think that proposed behaviour would be a valuable addition, then I agree it's probably something to change in `FinderLoader` as then it'd be consistent across both Rake tasks. It might be something to split into a separate "improve `FinderLoader#finder`" PR so as not to block this one.

Reply:
i would push for FinderLoader to load by slug, given the filename inconsistencies e.g. 
base path: find-licenses, filename: licence_transactions, 
base path: drug-device-alerts, filename: medical_safety_alerts 

and at least 3 more examples. 
But sure, i'll update this to use FinderLoader and put another PR out for FinderLoader

Reply:
Eeek 😱 I didn't realise the filenames/slugs were as inconsistent as that!

Yeah - definitely open a PR for improving `FinderLoader` then.